### PR TITLE
(GH-2478) Support module-install config when resolving modules

### PR DESCRIPTION
**Pull request by beechtom**
This updates the module resolver to support config from `module-install`
when resolving module dependencies. The resolver now configures a
`PuppetfileResolver::SpecSearchers::Configuration` object, setting the
proxy for git and Forge operations and the baseurl for Forge operations
if applicable.

!feature

* **Support `module-install` config when resolving modules**
  ([#2478](https://github.com/puppetlabs/bolt/issues/2478))

  The `bolt module add|install` commands and `Add|Install-BoltModule`
  cmdlets now support the `module-install` config option when resolving
  module dependencies.

---

### TODO

- [x] Mirror puppetfile-resolver 0.5.0 to artifactory
- [x] Update puppet-runtime to pull in puppetfile-resolver 0.5.0 https://github.com/puppetlabs/puppet-runtime/pull/405
- [x] Update gemspec to pin minimum version of puppetfile-resolver to `~> 0.5`, remove git gem from Gemfile

### CODE REVIEW

**Review by lucywyman — APPROVED** [review]
Woo! Verified this resolves modules from Artifactory, That both `proxy` and `forge.proxy` are respected, and that `HTTP_PROXY` and `HTTPS_PROXY` environment variables are respected.

**Review by lucywyman — APPROVED** [review]
Will wait for Heston to review the docs

File: documentation/bolt_installing_modules.md [hunk 1]
```diff
@@ -273,6 +273,62 @@ requirement.
 > documentation on [Specifying
 > versions](https://puppet.com/docs/puppet/latest/modules_metadata.html#specifying-versions).
 
+## Installing Forge modules from an alternate Forge
```
Comment (hestonhoffman) [inline documentation/bolt_installing_modules.md:276]:
```suggestion
## Install Forge modules from an alternate Forge
```

File: documentation/bolt_installing_modules.md [hunk 2]
```diff
@@ -273,6 +273,62 @@ requirement.
...
+
+You can configure Bolt to use a Forge other than the [Puppet
+Forge](https://forge.puppet.com) when it installs Forge modules. To configure
+Bolt to use an alternate Forge, set the `module-install` configuration option in
+either your project configuration file, `bolt-project.yaml`, or the default
+configuration file, `bolt-defaults.yaml`.
+
+To use an alterate Forge for installing Forge modules, set the `baseurl` key
```
Comment (hestonhoffman) [inline documentation/bolt_installing_modules.md:284]:
```suggestion
To use an alternate Forge for installing Forge modules, set the `baseurl` key
```

File: documentation/bolt_installing_modules.md [hunk 3]
```diff
@@ -273,6 +273,62 @@ requirement.
...
+  forge:
+    baseurl: https://forge.example.com
+```
+
+📖 **Related information**
+
+- [bolt-project.yaml options](bolt_project_reference.md#module-install)
+- [bolt-defaults.yaml options](bolt_defaults_reference.md#module-install)
+
+## Installing modules using a proxy
```
Comment (hestonhoffman) [inline documentation/bolt_installing_modules.md:299]:
```suggestion
## Install modules using a proxy
```

**Review by hestonhoffman — APPROVED** [review]
Couple small fixes, but LGTM!

**Pull request merged**